pkg/transform/kubernetes/populators_test.go

```diff
@@ -588,6 +588,53 @@ func TestPopulateEnvFrom(t *testing.T) {
 	}
 }
 
+func TestPopulateVolumes(t *testing.T) {
+	volumeClaims := []spec.VolumeClaim{{Name: "foo"}, {Name: "bar"}, {Name: "barfoo"}}
+	volumes := []api_v1.Volume{{Name: "foo"}}
+
+	// a volumeMount is defined but that is not there in volumeClaims
+	// neither it is in pod level volumes, so this should fail
+	failingContainers := []api_v1.Container{
+		{VolumeMounts: []api_v1.VolumeMount{{Name: "baz"}}},
+	}
+
+	if _, err := populateVolumes(failingContainers, volumeClaims, volumes); err == nil {
+		t.Errorf("should have failed but passed for volumeMount that" +
+			" does not exist.")
+	}
+
+	passingContainers := []api_v1.Container{
```

Comment:
I'm sorry, I'm nack on this.
Separating tests based on passing or failing makes it impossible to track the behavior being tested. We already have a structure in place, I don't understand why are we deviating from that.
I'm sorry if you feel this is bikeshedding, but this PR is deviating from the way things have been getting done for no good reason.
ping @kadel @cdrage